Robot_Adapter/Read/Elements/Node.cs

```diff
@@ -53,11 +55,11 @@ private List<Node> ReadNodes(List<string> ids = null)
                     bhomNodes.Add(bhomNode);
                 }
             }
-            else if (ids != null && ids.Count > 0)
+            else if (nodeIds != null && nodeIds.Count > 0)
```

Comment:
We are checking exactly the same thing here now twice. This could be changed to an simple `else`